src/main/java/de/tum/in/www1/artemis/domain/hestia/ProgrammingExerciseSolutionEntry.java

```diff
@@ -0,0 +1,122 @@
+package de.tum.in.www1.artemis.domain.hestia;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.DomainObject;
+import de.tum.in.www1.artemis.domain.ProgrammingExerciseTestCase;
+
+/**
+ * A ProgrammingExerciseSolutionEntry represents a single change in a file that a students has to make in order to pass the related test.
+ * It is structured similarly to a git diff entry.
+ * If it replaces existing code it will contain the previous code that it replaces otherwise previousCode will be null.
+ * If it is only removing existing code the code attribute will be null.
+ * If it encompasses the addition of an entire file, previousLine will be null.
+ * If it deletes an entire file, line will be null.
+ * previousLine and line will be different when there are other changes higher up in the file.
+ *
+ * Example:
+ * A print statement gets changed:
+ * SolutionEntry{
+ *         filePath = "<...>"
+ *     previousLine = 12
+ *     previousCode = System.out.println("Tset");
+ *             line = 12
+ *             code = System.out.println("Test");
+ * }
+ */
+@Entity
+@Table(name = "programming_exercise_solution_entry")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class ProgrammingExerciseSolutionEntry extends DomainObject {
+
+    @Column(name = "file_path")
+    private String filePath;
```

Comment:
Just something that crossed my mind:
Would it make sense to also have a previousFilePath? 
Could be interesting in exercises where you have to extract parts from one file into another file.
Not sure if that's within our implementation capabilities though.

Reply:
Detecting if a part of code was moved from one file to another is outside of our bachelor thesis, as this would be fairly difficult to accomplish. This would be represented by two separate SolutionEntries.